front-end/src/components/mapComponents/WeatherOverlay.tsx

```diff
@@ -20,17 +20,18 @@ const WeatherOverlay: React.FC = () => {
       .then((response) => response.json())
       .then((data) => {
         console.log("Weather API Response:", data); // Debugging: check response in console
-
+        console.log("Icon code:", data.current.weather[0].icon);
         if (data.current && data.current.weather && data.current.weather.length > 0) {
           setWeather({
             city_name: "Arlington", // Hardcoded city name
-            temp: Math.round(data.current.temp - 273.15), // Convert Kelvin to Celsius
-            feels_like: Math.round(data.current.feels_like - 273.15),
+            temp: Math.round(data.current.temp),
+            feels_like: Math.round(data.current.feels_like),
```

Comment:
Null checking please ? What if data.current return null ? then Math.round(null) will throw an exception ? How your app handle this exception ?